### ISSUE TITLE
Metadata reduction: Don't send out accurate timestamps

### ISSUE THREAD

**Issue by lucasmz-dev** [mjaakko/NeoStumbler#168]
It might be interesting to have a mode where privacy is taken more into account when it comes to contributing to a server.
More things could be added to it, but one thing I would like to propose is:

One nice thing I can think of is not sending accurate timestamps to the server, but rather, send out more summarized timestamps and don't send them in the same order. (e.g. all reports made in a single hour are all kept with the same timestamp of that hour, or same for day)

At the moment, location services get what is essentially a history of a user's location. This would give the user the option to reduce the resolution of the history here and make tracking harder.

This data doesn't seem to be very useful or authenticated, so just an average seems fine for most use. Even a full day of maximum resolution seems fine.

Discussed in beaconDB matrix: https://matrix.to/#/!rgcVclkrrFiLdpQftu:matrix.org/$cJoppJ1uvEzHR4uW8eqfJs6TjqdnUsfqsisa95S0y-Y?via=matrix.org&via=nitro.chat&via=nixos.dev
(channel does not have history enabled so I believe you need to be already in it, which mjaakko is)

**Comment by mjaakko** [comment]
Makes sense. I'll think about how a "privacy mode" could work. Few other things that need to be considered are speed and the direction of travel - those would allow to easily reconstruct the original path even if timestamps were truncated

**Comment by matt755** [comment]
I'm wondering if mapping all timestamps to their respective "at start of day" in the send reports function could help to ensure a little more privacy. Duplicates, however, would have to be identified and their timestamp counted up for a second as long as the BeaconDB lead dev doesn't update the server side processing "to something smarter", if I understood it correctly.

**Comment by joelkoen** [comment]
Reducing the accuracy of timestamps should not have any impact on beaconDB now :)

> I'm wondering if mapping all timestamps to their respective "at start of day" in the send reports function could help to ensure a little more privacy

Yeah, setting the timestamp to midnight in the device's local timezone sounds like a good idea.

**Comment by AtmosphericIgnition** [comment]
If BeaconDB no longer depends on accurate timestamps (and no other services I know of accepting submissions in the Ichnaea format), would it make sense to just have the one and only behavior be no accurate timestamps? That way, the user would be presented with less choices, as well a decreasing maintenance burden?
